Document tokenized From addresses for Emails (#727)
mautic/user-documentation: docs/channels/emails.rst

Reviewer: Add custom target.

 
```suggestion
.. _mailer as owner:

.. vale off
```

Mautic allows the use of tokens in Emails which gives the marketer the possibility to integrate a number of Contact fields in your Emails. These can be easily placed within your Emails and are automatically replaced with the appropriate text once sent.

> Check the :doc:`/configuration/variables` documentation for a list of all the available default fields.

Author: Added tokenized From addresses documentation based on PR #16060 which enables Contact field tokens in `FromEmailHelper.php` and `MailHelper.php`, implementing a sender resolution hierarchy: tokenized Advanced From > owner > plain Advanced From > system default (with token support).

Source: https://github.com/mautic/mautic/pull/16060

Tokenized From addresses
------------------------

Mautic allows you to use Contact field tokens in the From address and name fields. This makes Emails appear as though they're coming from a Contact-specific sender, such as their assigned Company.

You can use tokens in:

* The system-wide Email Settings **Name to send mail as** and **Email address to send mail from** fields
* The **From Name** and **From Address** fields on an individual Email's Advanced tab

For example, to send Emails from the Contact's Company:

.. code-block:: php

    {contactfield=companyname|Default Name}

>     {contactfield=companyemail|info@default.com}

Reviewer: Tidy up the indentations.

```suggestion
.. code-block:: php

   {contactfield=companyname|Default Name}
   {contactfield=companyemail|info@default.com}
```

The token format follows the standard Contact field syntax. You can include an optional default value after the ``|`` character. If Mautic can't resolve the token to a value, it uses the default value instead.

.. note::

>     The Contact field used in the Email address must contain a valid Email address. If using a custom field, ensure it's configured as an Email field type to guarantee proper validation.

Reviewer: ```suggestion
.. note::

   The Contact field used in the Email address must contain a valid Email address. If using a custom field, ensure it's configured as an Email field type to guarantee proper validation.
```

.. _sender-resolution-hierarchy:

Sender resolution hierarchy
^^^^^^^^^^^^^^^^^^^^^^^^^^^

When sending Emails, Mautic determines the From address using the following priority order:

1. **Tokenized Email Advanced From** - If the Email's Advanced tab has a From address with a Contact field token, and that token resolves to a valid value for the Contact, Mautic uses that address.
2. **Owner sender** - If **Mailer is owner** is enabled and the Contact has an assigned owner, Mautic uses the owner's Email address.
3. **Plain Email Advanced From** - If the Email's Advanced tab has a standard Email address, without tokens, Mautic uses that address.

> 4. **System default sender** - Mautic falls back to the system default from Email Settings. If the system default contains tokens, Mautic resolves them. If token resolution fails, Mautic uses the token's default value.

Reviewer: A couple of things here:

- Use # instead of number to follow our guidelines.
- Use active instead of passive voice.

```suggestion
#. **Tokenized Email Advanced From** - If the Email's Advanced tab has a From address with a Contact field token, and that token resolves to a valid value for the Contact, Mautic uses that address.
#. **Owner sender** - If **Mailer is owner** is enabled and the Contact has an assigned owner, Mautic uses the owner's Email address.
#. **Plain Email Advanced From** - If the Email's Advanced tab has a standard Email address, without tokens, Mautic uses that address.
#. **System default sender** - Mautic falls back to the system default from Email Settings. If the system default contains tokens, Mautic resolves them. If token resolution fails, Mautic uses the token's default value.
```

Mautic allows the use of tokens in Emails which gives the marketer the possibility to integrate a number of Contact fields in your Emails. These can be easily placed within your Emails and are automatically replaced with the appropriate text once sent.

Check the :doc:`/configuration/variables` documentation for a list of all the available default fields.

> Tokenized From addresses
> ------------------------

Reviewer: ```suggestion
.. vale off

Tokenized From addresses
------------------------

.. vale on
```

> Mautic allows you to use Contact field tokens in the From address and name fields. This makes Emails appear as though they're coming from a Contact-specific sender, such as their assigned Company.

Reviewer: ```suggestion
Mautic allows you to use Contact field tokens in the **From address** and **From name** fields. This makes Emails appear as though they're coming from a Contact-specific sender, such as their assigned Company.
```

You can use tokens in:

> * The system-wide Email Settings **Name to send mail as** and **Email address to send mail from** fields

Reviewer: ```suggestion
* The **Name to send mail as** and **Email address to send mail from** fields in the system-wide **Email Settings**
```

> * The **From Name** and **From Address** fields on an individual Email's Advanced tab

Reviewer: ```suggestion
* The **From Name** and **From Address** fields on an individual Email's **Advanced** tab
```

For example, to send Emails from the Contact's Company:

.. code-block:: php

   {contactfield=companyname|Default Name}
   {contactfield=companyemail|info@default.com}

The token format follows the standard Contact field syntax. You can include an optional default value after the ``|`` character. If Mautic can't resolve the token to a value, it uses the default value instead.

.. note::

   The Contact field used in the Email address must contain a valid Email address. If using a custom field, ensure it's configured as an Email field type to guarantee proper validation.

> .. _sender-resolution-hierarchy:

Reviewer: ```suggestion
.. _sender resolution hierarchy:
```

Sender resolution hierarchy
^^^^^^^^^^^^^^^^^^^^^^^^^^^

> When sending Emails, Mautic determines the From address using the following priority order:

Reviewer: ```suggestion
When sending Emails, Mautic determines the **From address** using the following priority order:
```

> #. **Tokenized Email Advanced From** - If the Email's Advanced tab has a From address with a Contact field token, and that token resolves to a valid value for the Contact, Mautic uses that address.

Reviewer: ```suggestion
#. **Tokenized Email Advanced From** - If the Email's **Advanced** tab has a **From address** with a Contact field token, and that token resolves to a valid value for the Contact, Mautic uses that address.
```

#. **Owner sender** - If **Mailer is owner** is enabled and the Contact has an assigned owner, Mautic uses the owner's Email address.

> #. **Plain Email Advanced From** - If the Email's Advanced tab has a standard Email address, without tokens, Mautic uses that address.

Reviewer: ```suggestion
#. **Plain Email Advanced From** - If the Email's **Advanced** tab has a standard Email address, without tokens, Mautic uses that address.
```

> #. **System default sender** - Mautic falls back to the system default from Email Settings. If the system default contains tokens, Mautic resolves them. If token resolution fails, Mautic uses the token's default value.

Reviewer: ```suggestion
#. **System default sender** - Mautic falls back to the system default from **Email Settings**. If the system default contains tokens, Mautic resolves them. If token resolution fails, Mautic uses the token's default value.
```

Check the :doc:`/configuration/variables` documentation for a list of all the available default fields.

.. vale off

Tokenized From addresses
------------------------

.. vale on

Mautic allows you to use Contact field tokens in the **From address** and **From name** fields. This makes Emails appear as though they're coming from a Contact-specific sender, such as their assigned Company.

You can use tokens in:

* The **Name to send mail as** and **Email address to send mail from** fields in the system-wide **Email Settings**
* The **From Name** and **From Address** fields on an individual Email's **Advanced** tab

For example, to send Emails from the Contact's Company:

.. code-block:: php

   {contactfield=companyname|Default Name}
   {contactfield=companyemail|info@default.com}

The token format follows the standard Contact field syntax. You can include an optional default value after the ``|`` character. If Mautic can't resolve the token to a value, it uses the default value instead.

.. note::

>    The Contact field used in the Email address must contain a valid Email address. If using a custom field, ensure it's configured as an Email field type to guarantee proper validation.

Reviewer: ```suggestion
   The Contact field used in the Email address must contain a valid Email address. If using a Custom Field, ensure it's configured as an Email field type to guarantee proper validation.
```

.. _sender resolution hierarchy:

Sender resolution hierarchy
^^^^^^^^^^^^^^^^^^^^^^^^^^^

When sending Emails, Mautic determines the **From address** using the following priority order:

#. **Tokenized Email Advanced From** - If the Email's **Advanced** tab has a **From address** with a Contact field token, and that token resolves to a valid value for the Contact, Mautic uses that address.

> #. **Owner sender** - If **Mailer is owner** is enabled and the Contact has an assigned owner, Mautic uses the owner's Email address.

Reviewer: Use active instead of passive voice. Address the "is enabled".

   .. note::

>       When enabled, **Mailer is owner** sends Emails from the Contact's owner, overriding system defaults and plain Email From addresses. However, tokenized From addresses on an Email's **Advanced** tab take precedence over the owner sender when the token resolves to a valid value. Every Contact owner's domain must have ``SPF`` and ``DKIM`` records. You can configure this setting for individual Emails, rather than globally. For more information, see :ref:`mailer as owner`. For the complete sender resolution hierarchy, see :ref:`Sender resolution hierarchy<sender resolution hierarchy>`.

Reviewer: @promptless-for-oss We've mentioned in the "Owner sender" description:

> If you enable **Mailer is owner** and the Contact has an assigned owner, Mautic uses the owner's Email address.

So, the first sentence in this note is a duplication. The last sentence where we link to sender resolution hirearchy also redundant because this note is exist in the same section. Also, we have the exaxt same note in the `configuration/settings.rst`. 

I'm not sure if we need this note admonition at all. If so, we might only want to mention the most important detail.
